yasa/evaluation.py

```diff
@@ -0,0 +1,649 @@
+"""
+YASA code for evaluating the agreement between two sleep-measurement systems.
+
+There are two levels of evaluating staging performance:
+- Comparing two hypnograms (e.g., human vs automated scorer)
+- Comparing summary sleep statistics between two scorers (e.g., PSG vs actigraphy)
+
+Analyses are modeled after the standardized framework proposed in Menghini et al., 2021, SLEEP.
+See the following resources:
+- https://doi.org/10.1093/sleep/zsaa170
+- https://sri-human-sleep.github.io/sleep-trackers-performance
+- https://github.com/SRI-human-sleep/sleep-trackers-performance
+"""
+import logging
+
+import numpy as np
+import pandas as pd
+import pingouin as pg
+from sklearn import metrics
+
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+from yasa.plotting import plot_hypnogram
+
+
+logger = logging.getLogger("yasa")
+
+__all__ = [
+    "EpochByEpochEvaluation",
+    "SleepStatsEvaluation",
+]
+
+
+class EpochByEpochEvaluation:
+    """
+    See :py:meth:`yasa.Hypnogram.evaluate`
+
+    Parameters
+    ----------
+    hypno_ref : :py:class:`yasa.Hypnogram`
+        Reference or ground-truth hypnogram.
+    hypno_test : :py:class:`yasa.Hypnogram`
+        The test or to-be-evaluated hypnogram.
+
+    Notes
+    -----
+    Many steps here are modeled after guidelines proposed in Menghini et al., 2021 [Menghini2021]_.
+    See https://sri-human-sleep.github.io/sleep-trackers-performance/AnalyticalPipeline_v1.0.0.html
+
+    References
+    ----------
+    .. [Menghini2021] Menghini, L., Cellini, N., Goldstone, A., Baker, F. C., & de Zambotti, M.
+                      (2021). A standardized framework for testing the performance of sleep-tracking
+                       technology: step-by-step guidelines and open-source code. Sleep, 44(2),
+                       zsaa170. https://doi.org/10.1093/sleep/zsaa170
+
+    Examples
+    --------
+    >>> import yasa
+    >>> hypno_a = yasa.simulate_hypnogram(tib=90, seed=8, scorer="RaterA")
+    >>> hypno_b = yasa.simulate_hypnogram(tib=90, seed=9, scorer="RaterB")
```

Comment:
I think this is not the best example because the two hypnograms are going to be very different. We could create a slightly-modified version of hypnogram A instead. More generally, I've been thinking that we could provide some pre-built example datasets in YASA, similar to Pingouin, sklearn, seaborn, etc. This would greatly simplify some docstring examples.

Reply:
Yes! I've been thinking the same. Actually I started writing some code for this a few months ago, based around [Pooch](https://www.fatiando.org/pooch). Not sure what you used for Pingouin, but Pooch is v cool and used by scikit-learn and mne for dataset management and downloading. I was gonna spend some development time on it before suggesting it, but maybe we could go ahead and start the conversation in YASA for `v0.8`.

Reply:
I actually stumbled upon Pooch this week, 100% in for using it! We can even start using it for v0.7